### PR TITLE
chore: restrict secret file permissions

### PR DESCRIPTION
**Pull request by driftluo**
Restrict the permissions of secret key files

### CODE REVIEW

File: util/app-config/src/configs/network.rs [hunk 1]
```diff
@@ -105,7 +116,14 @@ impl Config {
...
             .open(path)
-            .and_then(|mut file| file.write_all(&random_key_pair))
+            .and_then(|mut file| {
+                #[cfg(unix)]
+                {
+                    use std::os::unix::fs::PermissionsExt;
+                    file.set_permissions(fs::Permissions::from_mode(0o600))?;
+                }
+                file.write_all(&random_key_pair)
+            })
```
Comment (zhangsoledad) [inline util/app-config/src/configs/network.rs:126]: Maybe we can use `std::fs::set_permissions` here, this function works on Windows. rust said platform-specific behavior may change in the future, but that's future.

File: util/app-config/src/configs/network.rs [hunk 2]
```diff
@@ -105,7 +116,14 @@ impl Config {
             .create(true)
             .write(true)
             .open(path)
-            .and_then(|mut file| file.write_all(&random_key_pair))
+            .and_then(|mut file| {
+                #[cfg(unix)]
```
Comment (doitian) [inline util/app-config/src/configs/network.rs:120]: Does it also cover mac?

Comment (driftluo) [reply util/app-config/src/configs/network.rs:120]: `#[cfg(unix)]` contains mac

`cfg(target_os = "linux")` does not contain mac

File: util/app-config/src/configs/network.rs (hunk 1 above)
Comment (driftluo) [reply util/app-config/src/configs/network.rs:126]: Do we need to make separate settings for non-UNIX systems? I think the `read_only` restriction is not appropriate. What I want here is that only owner can access

File: util/app-config/src/configs/network.rs [hunk 3]
```diff
@@ -90,6 +90,17 @@ impl Config {
             Ok(file) => file,
             Err(_) => return Ok(None),
         };
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            if file.metadata()?.permissions().mode() != 0o600 {
+                ckb_logger::warn!(
```
Comment (quake) [inline util/app-config/src/configs/network.rs:97]: should we set the permission to 0o600 directly without warning log?

File: util/app-config/src/configs/network.rs [hunk 4]
```diff
@@ -90,6 +90,17 @@ impl Config {
             Ok(file) => file,
             Err(_) => return Ok(None),
         };
+        #[cfg(unix)]
+        {
+            use std::os::unix::fs::PermissionsExt;
+            if file.metadata()?.permissions().mode() != 0o600 {
```
Comment (doitian) [inline util/app-config/src/configs/network.rs:96]:
```suggestion
            if file.metadata()?.permissions().mode() & 0o177 != 0 {
```

0400 should also be accepted.

File: util/app-config/src/configs/network.rs [hunk 5]
```diff
@@ -105,7 +116,14 @@ impl Config {
             .create(true)
             .write(true)
             .open(path)
-            .and_then(|mut file| file.write_all(&random_key_pair))
+            .and_then(|mut file| {
+                #[cfg(unix)]
+                {
+                    use std::os::unix::fs::PermissionsExt;
+                    file.set_permissions(fs::Permissions::from_mode(0o600))?;
```
Comment (doitian) [inline util/app-config/src/configs/network.rs:123]:
```suggestion
                    file.set_permissions(fs::Permissions::from_mode(0o400))?;
```

File: util/app-config/src/configs/network.rs [hunk 6]
```diff
@@ -90,6 +90,26 @@ impl Config {
             Ok(file) => file,
             Err(_) => return Ok(None),
         };
+
+        let warn = |m: bool| {
+            if m {
+                ckb_logger::warn!(
+                    "Your network secret file's permission is not readonly, path: {:?}, \
```
Comment (yangby-cryptape) [inline util/app-config/src/configs/network.rs:97]: If a `secret_key` is `0444`, it's read only but it will cause a warning.

**Comment by doitian** [comment]
bors r=doitian,yangby-cryptape

**Pull request merged**